test/autocast_fp8.cpp

```diff
@@ -0,0 +1,172 @@
+        m1.add_instruction(migraphx::make_op("sub"), x, y);
+    }
+    run_pass(m1);
+
+    migraphx::module m2;
+    {
+        auto y_fp32 = m2.add_parameter("y", {migraphx::shape::float_type, {1}});
+        auto x_fp32 = m2.add_parameter("x", {migraphx::shape::float_type, {1}});
+        auto y_fp8  = m2.add_instruction(migraphx::make_op("convert", {{"target_type", migraphx::shape::fp8e4m3fnuz_type}}), y_fp32);
+        auto x_fp8  = m2.add_instruction(migraphx::make_op("convert", {{"target_type", migraphx::shape::fp8e4m3fnuz_type}}), x_fp32);
+        m2.add_instruction(migraphx::make_op("sub"), x_fp8, y_fp8);
+    }
+    EXPECT(m1 == m2);
+}
+
+// multiple inputs (of same type) to return
+TEST_CASE(autocast_fp8_3)
+{
+    migraphx::module m1;
+    {
+        auto x      = m1.add_parameter("x", {migraphx::shape::fp8e4m3fnuz_type, {1}});
+        auto y      = m1.add_parameter("y", {migraphx::shape::fp8e4m3fnuz_type, {1}});
+        auto sum    = m1.add_instruction(migraphx::make_op("add"), x, y);
+        auto diff   = m1.add_instruction(migraphx::make_op("sub"), x, y);
+        m1.add_return({sum, diff});
+    }
+    run_pass(m1);
+
+    migraphx::module m2;
+    {
+        auto y_fp32      = m2.add_parameter("y", {migraphx::shape::float_type, {1}});
+        auto x_fp32      = m2.add_parameter("x", {migraphx::shape::float_type, {1}});
+        auto y_fp8       = m2.add_instruction(migraphx::make_op("convert", {{"target_type", migraphx::shape::fp8e4m3fnuz_type}}), y_fp32);
+        auto x_fp8       = m2.add_instruction(migraphx::make_op("convert", {{"target_type", migraphx::shape::fp8e4m3fnuz_type}}), x_fp32);
+        auto sum_fp8     = m2.add_instruction(migraphx::make_op("add"), x_fp8, y_fp8);
+        auto diff_fp8    = m2.add_instruction(migraphx::make_op("sub"), x_fp8, y_fp8);
+        auto sum_fp32    = m2.add_instruction(migraphx::make_op("convert", {{"target_type", migraphx::shape::float_type}}), sum_fp8);
+        auto diff_fp32   = m2.add_instruction(migraphx::make_op("convert", {{"target_type", migraphx::shape::float_type}}), diff_fp8);
+        m2.add_return({sum_fp32, diff_fp32});
+    }
+    EXPECT(m1 == m2);
+}
+
+// multiple inputs (of different types) to return
+TEST_CASE(autocast_fp8_4)
+{
+    migraphx::module m1;
+    {
+        auto x1      = m1.add_parameter("x1", {migraphx::shape::fp8e4m3fnuz_type, {1}});
+        auto y1      = m1.add_parameter("y1", {migraphx::shape::fp8e4m3fnuz_type, {1}});
+        auto x2      = m1.add_parameter("x2", {migraphx::shape::float_type, {1}});
+        auto y2      = m1.add_parameter("y2", {migraphx::shape::float_type, {1}});
+        auto sum1    = m1.add_instruction(migraphx::make_op("add"), x1, y1);
+        auto sum2    = m1.add_instruction(migraphx::make_op("add"), x2, y2);
+        m1.add_return({sum1, sum2});
+    }
+    run_pass(m1);
+
+    migraphx::module m2;
+    {
+        auto x2          = m2.add_parameter("x2", {migraphx::shape::float_type, {1}});
```

Comment:
maybe change one of the shapes inside the test to be transposed. 
like `lens= {3, 4}, strides={1, 3}` 
Reason is that pass is using `with_type` somewhere. It should be fine but would be good for tests instead of having all the tests with same shapes.

Reply:
Done.